### ISSUE TITLE
valid-jsdoc with inheritDoc

### ISSUE THREAD

**Issue by thealjey** [eslint/eslint#4576]
Currently the following comment block is violating the `valid-jsdoc` rule.
```
/**
 * @inheritDoc
 */
```
Regardless of whether the method is documented in the parent class or not.
Is this a bug, a feature or something that will be changed in the future versions?
Can I use `@inheritDoc` with this rule right now?

**Comment by eslintbot** [comment]
@thealjey Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by gyandeeps** [comment]
I dont understand the issue here, can you please answer all the questions (above post)?

**Comment by thealjey** [comment]
Sorry, but it just seems like too much work for such a minor thing :smiley_cat: 
I simply removed the `@inheritDoc` and copied the whole comment block from the parent class.
I'll let whoever this bothers more than it does me to worry about this)))

**Comment by platinumazure** [comment]
It's still not clear what happened, what you're expecting (and why you're
expecting it) vs what actually happened. Version information is also
helpful in case you're on an old version and this is fixable with an
upgrade. Please answer the questions above.

**Comment by thealjey** [comment]
`JSDoc 3.4.0`
The rest is pretty much described in my original post (there's a documented method in a parent class and the same method in a child class with `/** @inheritDoc */`, that's it).
Sorry, but this is as much as I can bring myself to write right now.

**Comment by nzakas** [comment]
@thealjey We really need a clear example to work with to be able to both evaluate and potentially fix any bug. Keep in mind we get dozens of issues filed over the course of a week, so the more information you can provide, the more likely it is that someone will investigate further.

**Comment by cschuller** [comment]
@thealjey Specification by code example is a great way to explain things:

What @thealjey meant was something like this:

```javascript
'use strict';

/**
 * @constructor
 */
function Foo() {
    /**
     * @param {string} baz The first argument.
     * @return {void}
     */
    this.bar = function bar(baz) {};
}

/**
 * @constructor
 * @inherit {Foo}
 */
function MeAlsoFoo() {
    /**
     * @inheritDoc
     */
    this.bar = function bar(baz) {};
}
```

with a default ```valid-jsdoc``` config. will result in

```
  19:5  error  Missing JSDoc @returns for function  valid-jsdoc
  19:5  error  Missing JSDoc for parameter 'baz'    valid-jsdoc
```

I guess there is just no support for ```@inheritDoc``` within the ```valid-jsdoc``` rule. The implementation will not be easy, because currently the ```valid-jsdoc``` rule does not parse / know anything about types like ```{string}``` or ```{Foo}``` used within ```@inherit```

This is more a new feature of the rule itself than a bug.

**Comment by nzakas** [comment]
So in this case, we would just not check any JSDoc with inheritDoc?

**Comment by cschuller** [comment]
Yes, not checking sounds good.

Let's go for it, because there is already an "not-check" implementation for ```@override``` (#3629).

We can add an additional option ```noInheritDoc``` to disallow using of ```@inheritDoc```. If so then we also need to add another option ```noOverride``` to be consistent with current implementation.

I can do the implementation within the next 14 days (issue doesn't seem to be very critical).

**Comment by nzakas** [comment]
I don't think we need an option (since we don't have one for override either). It can just be the default behavior.

**Comment by cschuller** [comment]
Okay. I'm going to do this, give me two weeks. ;-)

**Comment by nzakas** [comment]
Awesome, thanks @cschuller

**Comment by alberto** [comment]
I think there is already support for this: https://github.com/eslint/eslint/blob/master/lib/rules/valid-jsdoc.js#L165

The correct syntax is `inheritdoc` (with lowercase d)

**Comment by nzakas** [comment]
Ah, I'm pretty sure that JSDoc is case insensitive when it comes to tags, so we probably just need to normalize the tag name to lowercase before that point.

**Comment by cschuller** [comment]
- The JSDoc documentation is using lowercase ```@inheritdoc``` [JSDoc](http://usejsdoc.org/tags-inheritdoc.html)
- The Google Closure Compiler expect a camelcase ```@inheritDoc``` [Google Developers](https://developers.google.com/closure/compiler/docs/js-for-compiler)
- WebStorm 11 likes to have a camelcase ```@inheritDoc```

I think a tag name normalization will do it. camel/lower case is then more a matter of styling.
